### PR TITLE
Main Readme

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,79 @@
-todo
+# Android Architecture Blueprints v2
+<p align="center">
+<img src="https://github.com/googlesamples/android-architecture/wiki/images/aab-logov2.png" alt="Android Architecture Blueprints"/>
+</p>
 
+Android Architecture Blueprints is a project to showcase different architectural approaches to developing Android apps. In its different branches you'll find the same app (a TODO app) implemented with small differences.
+
+In this master branch you'll find:
```
Comment (keyboardsurfer) [inline README.md]:
```suggestion
In this branch you'll find:
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,79 @@
...
+
+## Variations
+
+This project hosts each sample app in separate repository branches. For more information, see the `README.md` file in each branch.
+
+### Stable samples - Kotlin
+| Sample | Description |
+| ------------- | ------------- |
+| [master](/) | The base for the rest of the branches. Uses Kotlin, Architecture Components, coroutines, Data Binding, etc. |
+| [dagger](tree/dagger/) | A simple Dagger setup that uses `dagger-android` and removes the two flavors. |
```
Comment (keyboardsurfer) [inline README.md]:
```suggestion
| [dagger](tree/dagger/) | A basic Dagger setup that uses `dagger-android` and removes the two flavors. |
```

Comment (keyboardsurfer) [inline README.md]: Consider renaming `dagger` to `dagger-android`
One is easily capable of being used in a modularized app, the other one is not.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,79 @@
...
+
+### Old samples - Kotlin and Java
+
+Blueprints v1 had a collection of samples that are not maintained anymore, but can still be useful. See [all project branches](https://github.com/googlesamples/android-architecture/branches).
+
+## Why a to-do app?
+
+<img align="right" src="https://github.com/googlesamples/android-architecture/wiki/images/todoapp.gif" alt="A demo illustraating the UI of the app" width="288" height="512" style="display: inline; float: right"/>
+
+The app in this project aims to be simple enough that you can understand it quickly, but complex enough to showcase difficult design decisions and testing scenarios. For more information, see the [app's specification](https://github.com/googlesamples/android-architecture/wiki/To-do-app-specification).
```
Comment (keyboardsurfer) [inline README.md:42]:
```suggestion
The app in this project aims to be small enough to navigate quickly while being complex enough to showcase difficult design decisions and testing scenarios. For more information, see the [app's specification](https://github.com/googlesamples/android-architecture/wiki/To-do-app-specification).
```

**Review by keyboardsurfer — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -61,7 +61,7 @@ Clone the repository:
 ```
 git clone git@github.com:googlesamples/android-architecture.git
 ```
-This also checks out the master branch. If you want to change to a different sample: 
+This step checks out the master branch. If you want to change to a different sample:
```
Comment (pfmaggi) [inline README.md:64]: I would probalby re-phrase the previous text to:
Clone the repository and check out the master branch:

If you want to change to a different branch:

File: README.md [hunk 5]
```diff
@@ -22,10 +22,10 @@ This project hosts each sample app in separate repository branches. For more inf
 | Sample | Description |
 | ------------- | ------------- |
 | [master](/) | The base for the rest of the branches. Uses Kotlin, Architecture Components, coroutines, Data Binding, etc. |
-| [dagger](tree/dagger/) | A simple Dagger setup that gets rid of the two product flavors for testing. |
+| [dagger](tree/dagger/) | A simple Dagger setup that uses `dagger-android` and removes the two flavors. |
 | [usecases](tree/usecases/) | Adds a new domain layer that uses UseCases for business logic. |
 
-### Samples in progress - Kotlin
+### Samples in development - Kotlin
```
Comment (pfmaggi) [inline README.md:28]: Samples in "active" development -  Kotlin

File: README.md (hunk 3 above)
Comment (JoseAlcerreca) [reply README.md:42]: I'd prefer to keep this paragraph as it was written by a TW.

Comment (keyboardsurfer) [reply README.md:42]: SGTM

File: README.md [hunk 6]
```diff
@@ -1,5 +1,79 @@
...
 
+Android Architecture Blueprints is a project to showcase different architectural approaches to developing Android apps. In its different branches you'll find the same app (a TODO app) implemented with small differences.
+
+In this branch you'll find:
+*   Kotlin **[Coroutines](https://kotlinlang.org/docs/reference/coroutines-overview.html)** for background operations.
+*   A single-activity architecture, using the **[Navigation component](https://developer.android.com/guide/navigation/navigation-getting-started)** to manage fragment operations.
+*   A presentation layer that contains a fragment (View) and a **ViewModel** per screen (or feature).
+*   Reactive UIs using **LiveData** observables and **Data Binding**.
+*   A **data layer** with a repository and two data sources (local using Room and remote) that are queried with one-shot operations (no listeners or data streams).
+*   Two **product flavors**, `mock` and `prod`, [to ease development and testing](https://android-developers.googleblog.com/2015/12/leveraging-product-flavors-in-android.html) (except in the Dagger branch)
```
Comment (manuelvicnt) [inline README.md]: This sentence misses a full point

**Review by manuelvicnt — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (JoseAlcerreca) [reply README.md]: Done

**Pull request merged**